MetroidPrimeInterface.py

```diff
@@ -186,8 +186,19 @@ def __init__(self, logger) -> None:
     def give_item_to_player(self, item_id: int, new_amount: int, new_capacity: int, ignore_capacity: bool = False):
         """Gives the player an item with the specified amount and capacity"""
         if item_id in custom_charge_id_to_beam:
-            charge_beam = custom_charge_id_to_beam[item_id]
-            self.set_progressive_beam_charge_state(charge_beam, True)

```

Comment:
Could you encapsulate this all inside the  existing function instead? It seems like there could probably be a way to reduce a lot of this down into a similarly sized function

Reply:
I think we'd also want to reimplement the `get_progressive_beam_charge_state` function and remove the `"PROGRESSIVE_BEAM_ADDRESS"`

Reply:
I will edit the PR with the changes